src/common/userlist.c

```diff
@@ -130,8 +130,12 @@ userlist_add_hostname (struct session *sess, char *nick, char *hostname,
 			g_free (user->hostname);
 			user->hostname = g_strdup (hostname);
 		}
-		if (!user->realname && realname && *realname)
+		if (realname && (!user->realname || strcmp(user->realname, realname)))
+		{
+			do_rehash = TRUE;
```

Comment:
Is `do_rehash` needed? That is for elements that show up in the userlist.

Reply:
I was going off nearby code so I thought it did but now I check it doesn't. Removed.